Core/BookmarkOrFolder.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  BookmarkOrFolder.swift
+//  DuckDuckGo
+//
+//  Copyright © 2022 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+public class BookmarkOrFolder {
+    public let name: String
+    public let type: String
```

Comment:
No problem!  I've done it as an enum, hope that's what you had in mind?  Just FYI this class is the Mac app version with the only difference being on Mac it was a struct (on iOS it is a class since I'm mutating / continually adding to the children)

Do you like the enum approach better or more consistency with MacOS (or maybe this doesn't matter til later or not at all?)

Thanks!